### DISCUSSION TITLE
How to change graphical syntax from VSM operations

### DISCUSSION THREAD

**Discussion by t-chiang** [eclipse-sirius/sirius-desktop#478] [Q&A]
Hello all,

Is there a way to change graphical syntax of my editor using the VSM specifier instead of using an external Java action? Going through some of the documentation there don't seem to be any operations that facilitate access semantic elements of the model as type DNode or DSemanticContainer to change their style (border, color, etc). I'm trying to create an operation that will go through referenced elements in the model and change their style, but nothing else about them semantically.

**Comment by lredor** [comment]
Hi,
The graphical aspect is directly managed through the VSM with the "Styles" and "[Conditional Styles](https://eclipse.dev/sirius/doc/specifier/diagrams/Diagrams.html#conditional_styles)". The `predicate expression` of a conditional style is precisely there to use semantic notions to condition the graphic style.

For example, in this diagram, the abstract classes are in red, bold and with large border:
![image](https://github.com/user-attachments/assets/f1247f70-99ef-425b-9733-f4bca970a8b3)
![image](https://github.com/user-attachments/assets/cddb589c-a737-4f16-9028-53cacb300369)
